.github/workflows/doc.yml

```diff
@@ -19,7 +19,6 @@ jobs:
         with:
           profile: minimal
           override: true
-      - name: Install cargo make
-        run: cargo install cargo-make
+      - uses: davidB/rust-cargo-make@v1
```

Comment:
What benefit do we gain from using this custom version?

Reply:
It is now the new "Gold standard" for this use case, it's the most used one if you look at github actions pertaining to rust installation: https://github.com/marketplace?category=&type=actions&verification=&query=sort%3Apopularity-desc+rust

Second most used rust github crate

Reply:
Are you referring to [rust-cargo](https://github.com/marketplace/actions/rust-cargo) or [rust-cargo-make](https://github.com/marketplace/actions/rust-cargo-make)? The latter one doesn't seem to be all the popular.

Reply:
Indeed in this PR I have added rust-cargo-make action. Taken this advice from the documentation of cargo-make here: https://github.com/sagiegurari/cargo-make/blob/d3c74835774b957cccba6ad3be057545d24729b6/README.md?plain=1#L3165

Reply:
Not widely used indeed but speeds up the CI where the installation / compilation of the cargo-make crate takes some time. In case you prefer to use the install method I can go back to using it.

Reply:
I guess it depends on how much of a difference it makes in CI. How much time are we saving by going to `rust-cargo-make`?